src/OrleansProviders/Streams/Common/SimpleCache/SimpleQueueCache.cs

```diff
@@ -249,6 +246,8 @@ internal bool TryGetNextMessage(SimpleQueueCacheCursor cursor, out IBatchContain
             {
                 AdvanceCursor(cursor, cursor.Element.Previous);
             }
+
+            batch = cursor.Element?.Value.Batch;
             return true;
```

Comment:
return (batch != null)?

Reply:
Yeah, that would be correct. Oddly not doing that is harmless, merely causing an extra iteration of the loop, but it is still incorrect, so I'll fix it.